HSDS/Writer/Writer.cpp

```diff
@@ -508,33 +508,56 @@ ACE_TMAIN(int argc, ACE_TCHAR *argv[])
-  HsdsResource<HSDS3::RequiredDocument> required_document_hsds_resource(application, ws);
-  HsdsResource<HSDS3::Contact> contact_hsds_resource(application, ws);
-  HsdsResource<HSDS3::OrganizationIdentifier> organization_identifier_hsds_resource(application, ws);
-  HsdsResource<HSDS3::Attribute> attribute_hsds_resource(application, ws);
-  HsdsResource<HSDS3::Metadata> metadata_hsds_resource(application, ws);
-  HsdsResource<HSDS3::MetaTableDescription> meta_table_description_hsds_resource(application, ws);
-  HsdsResource<HSDS3::Taxonomy> taxonomy_hsds_resource(application, ws);
-  HsdsResource<HSDS3::TaxonomyTerm> taxonomy_term_hsds_resource(application, ws);
-  HsdsResource<HSDS3::Address> address_hsds_resource(application, ws);
-
-  ws.start(true);
+  HsdsResource<HSDS3::Service> service_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::Phone> phone_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::Schedule> schedule_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::ServiceArea> service_area_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::ServiceAtLocation> service_at_location_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::Location> location_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::Language> language_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::Organization> organization_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::Funding> funding_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::Accessibility> accessibility_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::CostOption> cost_option_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::Program> program_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::RequiredDocument> required_document_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::Contact> contact_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::OrganizationIdentifier> organization_identifier_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::Attribute> attribute_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::Metadata> metadata_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::MetaTableDescription> meta_table_description_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::Taxonomy> taxonomy_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::TaxonomyTerm> taxonomy_term_hsds_resource(application, webserver);
+  HsdsResource<HSDS3::Address> address_hsds_resource(application, webserver);
+
+  webserver.start(false);
+
+  GuardWrapper wrapper;
+  DDS::WaitSet_var ws = new DDS::WaitSet;
+  ws->attach_condition(wrapper.guard());
+  DDS::ConditionSeq active;
+
+  bool keep_going = true;
+  const DDS::Duration_t infinite = { DDS::DURATION_INFINITE_SEC, DDS::DURATION_INFINITE_NSEC };
+  while (keep_going) {
+    DDS::ReturnCode_t error = ws->wait(active, infinite);
```

Comment:
```suggestion
    const DDS::ReturnCode_t error = ws->wait(active, infinite);
```
And maybe log the error code in the error message. Other than that, it's ready to go.